src/libraries/QuoterRevert.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {ParseBytes} from "@uniswap/v4-core/src/libraries/ParseBytes.sol";
+
+library QuoterRevert {
```

Comment:
I find this library a little difficult to reason about because all these functions seem to be doing some of the same things but at various lifecycles of a quote (various parsing, and various reverting) but its unclear when to use what.

some suggestions??

- does `validateRevertReason` need to be its own function? Seems like its just used in this file. Can you collapse it?
- for all of the reverting code paths can we decide if we want to use the revert logic in assembly or not? `revertQuote` and `bubbleReason` are not consistent with the reverting info.
- can you add documentation for whats the difference between those two functions? its unclear.. it seems like `revertQuote` could just use `bubbleReason`?
- do we need `parseAmountUnspecified as its own function as well? it seems like its just used in parseReturnData? Maybe we can change parseReturnData -> parseAmountUnspecified to be more clear since it doesnt parse any other kind of data?